cli: add option to list only conflicted branches

Kintaro opened the pull request:
As requested in #1471, I added a new flag for `jj branch list` to only show branches that are conflicted.

Adds a unit test to check for listing only conflicted branches and regenerates the cli output to incorporate the new flag.

Closes #1471

reformat

<!--
There's no need to add anything here, but feel free to add a personal message.
Please describe the changes in this PR in the commit message(s) instead, with
each commit representing one logical change. Address code review comments by
rewriting the branch rather than adding commits on top. Use force-push when
pushing the updated branch (`jj git push` does that automatically when you
rewrite a branch). Merge the PR at will once it's been approved. See
https://github.com/martinvonz/jj/blob/main/docs/contributing.md for details.
Note that you need to sign Google's CLA to contribute.
-->

# Checklist

If applicable:
- [X] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [X] I have added tests to cover my changes

yuja commented on cli/tests/test_branch_command.rs:
```diff
@@ -1348,6 +1348,80 @@ fn test_branch_list_tracked() {
+    main: sptzoqmo 7b33f629 commit 1
+      @origin: sptzoqmo 7b33f629 commit 1
+    "###);
+    insta::assert_snapshot!(get_conflicted_branch_output(&test_env, &repo_path), @r###"
```
nit: since this is the test for `jj branch list` command, I would inline the command here.

Kintaro replied:
Makes sense.

yuja commented on cli/tests/test_branch_command.rs:
```diff
@@ -1348,6 +1348,80 @@ fn test_branch_list_tracked() {
+            "refs/heads/feature2",
+        ],
+    );
+    test_env.add_config("git.auto-local-branch = false");
```
nit: perhaps, these setup codes are copied from somewhere, and can be simplified?

If we don't care about the remote branches, the easiest way to create conflicts is:
```
jj new 'root()' -m a
jj new 'root()' -m b
jj branch create foo -rREV_A
jj branch create foo -rREV_B --at-op=@-
```

Kintaro replied:
Yes, they were partially copied from another branch test. The smaller setup looks good, I'll change it to that.

yuja reviewed (approved):
lgtm, thanks!

Kintaro commented:
I have updated it, but had to change the way I was retrieving the output, otherwise `jj_cmd_success` complained due to

```
stdout=
bar: kkmpptxz 06a973bc (empty) b
foo (conflicted):
  + rlvkpnrz d8d5f980 (empty) a
  + kkmpptxz 06a973bc (empty) b

stderr="Concurrent modification detected, resolving automatically.\n"
```

I hope that's ok. If there's a better way how this should be handled in `jj`, I will gladly change it.

yuja commented:
> I have updated it, but had to change the way I was retrieving the output, otherwise `jj_cmd_success` complained due to

The common pattern is `let (stdout, stderr) = jj_cmd_ok(..)`. You can also insert some command (like `jj status`) before `get_branch_output()`. The diverged operations will be merged there.

Kintaro commented:
Oh right, forgot about the internal operations log. I have added a `jj status` call and reverted the `branch list --all` and `branch list --conflicted` to use `jj_cmd_success`.
